File: web/pwa/lib/render_descriptor_v1_cli.js
#!/usr/bin/env node
// CLI wrapper for render_descriptor_v1: reads an assistant_turn_v1
// response JSON object from stdin, writes the render-descriptor-v1
// JSON to stdout. Used by tests/unit/clients/render_descriptor_canary_test.go.
'use strict';

const { renderToDescriptorV1 } = require('./render_descriptor_v1.js');

let raw = '';
process.stdin.setEncoding('utf8');
process.stdin.on('data', (chunk) => {
  raw += chunk;
});
process.stdin.on('end', () => {
  try {
    const input = JSON.parse(raw);
    const out = renderToDescriptorV1(input);
    process.stdout.write(JSON.stringify(out));
  } catch (err) {
    process.stderr.write(String((err && err.message) ? err.message : err));
    process.exit(2);
  }
});

File: clients/mobile/assistant/tool/render_descriptor_v1_cli.dart
// CLI wrapper for render_descriptor_v1: reads a spec 069 assistant_turn_v1
// response JSON object from stdin, writes the canonical
// render-descriptor-v1 JSON to stdout. Used by the cross-language
// renderer canary at
// `tests/unit/clients/render_descriptor_canary_test.go` (TP-073-03).
//
// Run from the package root:
//   dart run tool/render_descriptor_v1_cli.dart < input.json

import 'dart:convert';
import 'dart:io';

import 'package:smackerel_assistant/core/render_descriptor_v1.dart';

Future<void> main(List<String> args) async {
  final String raw = await utf8.decoder.bind(stdin).join();
  try {
    final Object? decoded = jsonDecode(raw);
    if (decoded is! Map<String, dynamic>) {
      stderr.write('render_descriptor_v1_cli: stdin must be a JSON object');
      exitCode = 2;
      return;
    }
    final Map<String, dynamic> descriptor = renderToDescriptorV1(decoded);
    stdout.write(jsonEncode(descriptor));
  } catch (e) {
    stderr.write(e.toString());
    exitCode = 2;
  }
}
